### PR TITLE
PollingControlProxy refactoring 

### PR DESCRIPTION
**Pull request by daschuer**
This is the first PR with non conflicting changes after my original attempt, waiting here: 
https://github.com/daschuer/mixxx/tree/controlproxylt2

This replaces some ControlProxies with PollingControlProxies and avoids heap allocation

### CODE REVIEW

**Review by Swiftb0y — COMMENTED** [review]
Thanks. I have a few questions. Lgtm otherwise.

File: src/soundio/soundmanager.cpp [hunk 1]
```diff
@@ -51,7 +51,9 @@ SoundManager::SoundManager(UserSettingsPointer pConfig,
           m_config(this),
           m_pErrorDevice(nullptr),
           m_underflowHappened(0),
-          m_underflowUpdateCount(0) {
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/soundio/soundmanager.cpp:54]: Ups, no, not at all. 
This slipped in die to rebasing my original version. 

For this reason I have reverted all conflicts, from my branch where every polling CO was replaced. 

I need to redo this.

File: src/recording/recordingmanager.h [hunk 2]
```diff
@@ -60,7 +60,6 @@ class RecordingManager : public QObject {
     // name of the first split but with a suffix.
     void splitContinueRecording();
     void warnFreespace();
-    ControlProxy* m_recReady;
     ControlObject* m_recReadyCO;
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/recording/recordingmanager.h:63]: I think this was a left over from a earlier version where the two COs where used to control if a value changed signal is send to the transmitting object.

This should be obsolete now.

But anyway, it deserves a separate commit.

File: src/soundio/soundmanager.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (daschuer) [reply src/soundio/soundmanager.cpp:54]: Re-added, see the line below

**Comment by daschuer** [comment]
Now with an improved commit message.

File: src/engine/channels/enginechannel.h [hunk 3]
```diff
@@ -1,9 +1,10 @@
 #pragma once
 
 #include "control/controlproxy.h"
```
Comment (ferranpujolcamins) [inline src/engine/channels/enginechannel.h:3]: I believe this include can be removed now.

**Comment by ferranpujolcamins** [comment]
LGTM. Thank you!

**Review by Swiftb0y — APPROVED** [review]
lgtm too

**Review by ferranlala — APPROVED** [review]

**Comment by daschuer** [comment]
Done.

**Comment by daschuer** [comment]
Rebased to fix CI issue

**Comment by ferranpujolcamins** [comment]
@daschuer do you want to work more on this? Or can it be merged?

**Comment by daschuer** [comment]
This is ready for merge.

**Pull request merged**